### ISSUE TITLE
Modal - Avoid the use of implicit headings - (2036570034)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#7802]
### Violation: 
 Avoid the use of implicit headings 

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) 

### Severity: 
 6 

### Media Type: 
 Page Structure 

--- 
### Areas for Remediation: 
- Title: Modal - Avoid the use of implicit headings - (2036570034) 
- Module: Modal 
**Issue**
There is content that functions as a heading but does not use heading markup. Examples include:
- Small scale modal
- Medium scale modal
- Large scale modal
- Destructive action modal, etc.

**User Impact**
Screen reader users will have difficulty efficiently navigating and gaining an accurate overview of the page.

**Code Reference** 
 ```
<div slot="header" id="modal-title">Small scale modal</div>
 ``` 
**Suggestion**
Avoid the use of implicit headings. Text that visually appears as a heading for a section of content must use ``` <h1> ```, ``` <h2> ```, etc. elements. Additionally, ensure the level of the heading accurately reflects the element&#039;s position in the visual hierarchy.

**Compliant Code Example** 
 ```
<h2 slot="header" id="modal-title">Small scale modal</h2>
 ```
 
--- 
### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by macandcheese** [comment]
For now, this is a configuration issue as the slotted content can by any element. If we proceed with @driskull suggestion to mode Modal -> Dialog and either wrap or accept a slotted Panel, the heading can be provided via property or the Panel's heading-content slot (which, would also be configuration by user).

Seems safe to close this one.

**Comment by geospatialem** [comment]
Closing per https://github.com/Esri/calcite-design-system/issues/7802#issuecomment-1781439162.

For the `dialog` component context (deprecating `modal`), refer to https://github.com/Esri/calcite-design-system/issues/7886.
